src/test/java/org/takes/rs/RsVelocityTest.java

```diff
@@ -54,4 +56,23 @@ public void buildsTextResponse() throws IOException {
         );
     }
 
+    /**
+     * RsVelocity should close template's InputStream after serving response.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void closesTemplateInputStream() throws IOException {
+        final String template = "hello, world!";
+        final MockInputStream templateStream = new MockInputStream(template);
+        MatcherAssert.assertThat(
+            IOUtils.toString(
+                new RsVelocity(
+                    templateStream,
+                    Collections.<CharSequence, Object>emptyMap()
+                ).body()
+            ),
+            Matchers.equalTo(template)
+        );
+        MatcherAssert.assertThat(templateStream.isClosed(), Matchers.is(true));
```

Comment:
@carlosmiranda As you can see here https://github.com/Happy-Neko/takes/blob/49f4c47b87a3c9a5ae0e2fec343f1ebee6b01798/src/main/java/org/takes/rs/RsVelocity.java#L146 `InputStream` under test is passed to external library (Velocity) and we want to test whether it was closed here or not. I don't think it will be easy to mock it with Mockito since we have no control on what methods are called within external code. May be it is possible to use partial mocks to spy on real `InputStream` object. But extending real `InputStream` with required behavior feels like cleaner solution.